src/PAModel/MsAppTest.cs

```diff
@@ -211,41 +211,54 @@ public static bool StressTest(string pathToMsApp)
                 return false;
             }
 
-            
-
             return true;
         }
 
         public static bool Compare(string pathToZip1, string pathToZip2, TextWriter log)
         {
+            ErrorContainer errorContainer = new ErrorContainer();
+
             var c1 = ChecksumMaker.GetChecksum(pathToZip1);
             var c2 = ChecksumMaker.GetChecksum(pathToZip2);
             if (c1.wholeChecksum == c2.wholeChecksum)
             {
                 return true;
             }
 
-            // If there's a checksum mismatch, do a more intensive comparison to find the difference.
-#if DEBUG
             // Provide a comparison that can be very specific about what the difference is.
             var comp = new Dictionary<string, byte[]>();
-            DebugChecksum(pathToZip1, log, comp, true);
-            DebugChecksum(pathToZip2, log, comp, false);
 
-            foreach (var kv in comp) // Remaining entries are errors.
+            CompareChecksums(pathToZip1, log, comp, true, errorContainer);
+            CompareChecksums(pathToZip2, log, comp, false, errorContainer);
+
+            return false;
+        }
+
+        // Overload with ErrorContainer
+        public static bool Compare(string pathToZip1, string pathToZip2, TextWriter log, ErrorContainer errorContainer)
```

Comment:
For the overload, we want to avoid code duplication where possible. I meant something more along the lines of creating a new `Compare` with an `ErrorContainer` (which you have done) and then changing the previous `Compare` function to look something like:

```
public static bool Compare(string pathToZip1, string pathToZip2, TextWriter log)
{
   ErrorContainer errorContainer = new ErrorContainer();
   return Compare(pathToZip1, pathToZip2, log, errorContainer);
}
```

Now your meaningful logic is contained in a single location, but you can continue to call the old method definition and it will be handled as expected.